test/WebJobs.Script.Tests/Rpc/RpcInitializationServiceTests.cs

```diff
@@ -46,78 +68,130 @@ public async Task RpcInitializationService_Initializes_RpcServerAndChannels_Plac
-                Assert.DoesNotContain("testserver", testRpcServer.Uri.ToString());
-                await testRpcServer.ShutdownAsync();
-            }
-            finally
-            {
-                TestHelpers.DeleteTestFile(offlineFilePath);
-            }
+            mockEnvironment.Setup(p => p.GetEnvironmentVariable(EnvironmentSettingNames.FunctionsLogsMountPath)).Returns(@"d:\test");
+            mockEnvironment.Setup(p => p.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteInstanceId)).Returns("1234");
+            mockEnvironment.Setup(p => p.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsitePlaceholderMode)).Returns("0");
+            mockEnvironment.Setup(p => p.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName)).Returns(string.Empty);
+
+            _rpcInitializationService = new RpcInitializationService(_optionsMonitor, mockEnvironment.Object, testRpcServer, _mockLanguageWorkerChannelManager.Object, _loggerFactory);
+            await _rpcInitializationService.StartAsync(CancellationToken.None);
+            _mockLanguageWorkerChannelManager.Verify(m => m.InitializeChannelAsync(LanguageWorkerConstants.PythonLanguageWorkerName), Times.Never);
+            _mockLanguageWorkerChannelManager.Verify(m => m.InitializeChannelAsync(LanguageWorkerConstants.JavaLanguageWorkerName), Times.Never);
+            Assert.Contains("testserver", testRpcServer.Uri.ToString());
+            await testRpcServer.ShutdownAsync();
+        }
+
+        [Fact]
+        public async Task RpcInitializationService_Initializes_RpcServerAndChannels_AppLevel_WorkerRuntime_Set_NoPlaceHolderMode()
+        {
+            IRpcServer testRpcServer = new TestRpcServer();
+            var mockEnvironment = new Mock<IEnvironment>();
+            mockEnvironment.Setup(p => p.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName)).Returns(LanguageWorkerConstants.NodeLanguageWorkerName);
+
+            _rpcInitializationService = new RpcInitializationService(_optionsMonitor, mockEnvironment.Object, testRpcServer, _mockLanguageWorkerChannelManager.Object, _loggerFactory);
+            _rpcInitializationService.AddSupportedAppLeveRuntime(LanguageWorkerConstants.NodeLanguageWorkerName);
+            await _rpcInitializationService.StartAsync(CancellationToken.None);
+            _mockLanguageWorkerChannelManager.Verify(m => m.InitializeChannelAsync(LanguageWorkerConstants.NodeLanguageWorkerName), Times.Once);
+            Assert.Contains("testserver", testRpcServer.Uri.ToString());
+            await testRpcServer.ShutdownAsync();
         }
 
         [Fact]
-        public async Task RpcInitializationService_Initializes_WorkerRuntime_Set()
+        public async Task RpcInitializationService_Initializes_RpcServerAndChannels_AppLevel_WorkerRuntime_Set_PlaceHolderMode()
```

Comment:
Add another test, set workerRuntime to Python, not in placeholder mode verify ptyhon worker does not get initialized